frame/staking/src/voter_bags.rs

```diff
@@ -899,11 +944,137 @@ mod tests {
 				Staking::update_ledger(&controller, &ledger);
 				Staking::do_rebag(voter_id);
 
-				balance *= 2;
+				// Increase balance to the next threshold.
+				balance += 10;
 			}
 
 			let have_voters: Vec<_> = VoterList::<Test>::iter().map(|node| node.voter.id).collect();
 			assert_eq!(voters, have_voters);
 		});
 	}
+
+	/// This tests that we can `take` x voters, even if that quantity ends midway through a list.
+	#[test]
+	fn take_works() {
+		ExtBuilder::default().validator_pool(true).build_and_execute(|| {
+			// initialize the voters' deposits
+			let mut balance = 0; // This will be 10 on the first loop iteration because 0 % 3 == 0
+			for (idx, voter_id) in GENESIS_VOTER_IDS.iter().enumerate() {
```

Comment:
why are we doing this? `GENESIS_VOTER_IDS` should already be in the correct bag.

Reply:
I am thinking its easier to read the test when you can see up front how they are distributed in the bags. I need to double check how the genesis voters are distributed, but the scheme here guarantees that the last node `take`n is mid-bag, which is an edge case we talked about explicitly testing

Reply:
I see your point, but when you have 100+ tests, it is quite verbose to setup things manually for each tests. Instead, it is more reasonable to *assume* some fixed genesis stakers (and the programmer should simply have this in their mind) and write all tests preferably based on that.